src/AuthProvider.tsx

```diff
@@ -104,23 +108,38 @@ export const AuthProvider = (props: AuthProviderProps): JSX.Element => {
         onSignoutPopup,
 
         implementation: UserManagerImpl = defaultUserManagerImpl,
-        ...userManagerProps
+        ...userManagerSettings
     } = props;
 
-    const [userManager] = useState(() => UserManagerImpl && new UserManagerImpl(userManagerProps));
+    const [userManager] = useState(() => UserManagerImpl && new UserManagerImpl(userManagerSettings));
     const [state, dispatch] = useReducer(reducer, initialAuthState);
     const userManagerContext = useMemo(
-        () => ({
-            settings: userManager?.settings ?? { ...userManagerProps },
-            ...(Object.fromEntries(
+        () => Object.assign(
+            { settings: userManager?.settings ?? userManagerSettings },
+            Object.fromEntries(
                 userManagerContextKeys.map((key) => [
                     key,
                     userManager
                         ? userManager[key].bind(userManager)
                         : unsupportedEnvironment(key),
                 ]),
-            ) as Pick<UserManager, typeof userManagerContextKeys[number]>),
-        }),
+            ) as Pick<UserManager, typeof userManagerContextKeys[number]>,
+            Object.fromEntries(
+                navigatorKeys.map((key) => [
+                    key,
+                    userManager
+                        ? async (...args: never[]) => {
+                            dispatch({ type: "NAVIGATOR_INIT", method: key });
+                            try {
+                                return await userManager[key](...args);
+                            } finally {
+                                dispatch({ type: "NAVIGATOR_CLOSE" });
+                            }
+                        }
+                        : unsupportedEnvironment(key),
+                ]),
+            ) as Pick<UserManager, typeof navigatorKeys[number]>,
+        ),
         [userManager],
```

Comment:
depends new additionally on `userManagerSettings`

Reply:
Only trivially, since it's only used server-side for the initial render. We don't want to include it as a dependency because that value changes every render since it's defined via rest properties.

Reply:
I worked around this by using the provided settings to create a fake UserManager in the `useState` hook.